src/Components/Aspire.Azure.Messaging.EventHubs/AspireEventHubsExtensions.cs

```diff
@@ -0,0 +1,241 @@
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="AzureMessagingEventHubsSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureClientBuilder">An optional method that can be used for customizing the <see cref="IAzureClientBuilder{TClient, TOptions}"/>.</param>
+    /// <remarks>Reads the configuration from "Aspire:Azure:Messaging:EventHubs:{TClient}" section, where {TClient} is the type of Event Hubs client being configured, i.e. EventProcessorClient.</remarks>
+    /// <exception cref="InvalidOperationException">Thrown when neither <see cref="AzureMessagingEventHubsSettings.ConnectionString"/> nor <see cref="AzureMessagingEventHubsSettings.Namespace"/> is provided.</exception>
+    public static void AddKeyedAzureEventHubConsumerClient(
+        this IHostApplicationBuilder builder,
+        string name,
+        Action<AzureMessagingEventHubsSettings>? configureSettings = null,
+        Action<IAzureClientBuilder<EventHubConsumerClient, EventHubConsumerClientOptions>>? configureClientBuilder = null)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        string configurationSectionName = EventHubConsumerClientComponent
+            .GetKeyedConfigurationSectionName(name, DefaultConfigSectionName +
+                                                    nameof(EventHubConsumerClient));
+
+        new EventHubConsumerClientComponent()
+            .AddClient(builder, configurationSectionName, configureSettings,
+                configureClientBuilder, connectionName: name, serviceKey: name);
+    }
+}
+
+internal abstract class EventHubsComponent<TClient, TClientOptions> :
+    AzureComponent<AzureMessagingEventHubsSettings, TClient, TClientOptions>
+    where TClientOptions: class
+    where TClient : class
+{
+    protected override IHealthCheck CreateHealthCheck(TClient client, AzureMessagingEventHubsSettings settings)
+        => throw new NotImplementedException();
+
+    protected override void BindSettingsToConfiguration(AzureMessagingEventHubsSettings settings, IConfiguration config)
+    {
+        config.Bind(settings);
+    }
+
+    protected override bool GetHealthCheckEnabled(AzureMessagingEventHubsSettings settings)
+        => false;
+
+    protected override TokenCredential? GetTokenCredential(AzureMessagingEventHubsSettings settings)
+        => settings.Credential;
+
+    protected override bool GetTracingEnabled(AzureMessagingEventHubsSettings settings)
+        => settings.Tracing;
+
+    protected static string GenerateClientIdentifier(AzureMessagingEventHubsSettings settings)
+    {
+        // configure processor identifier
+        var slug = Guid.NewGuid().ToString().Substring(24);
+        var identifier = $"{Environment.MachineName}-{settings.EventHubName}-" +
```

Comment:
Is putting the MachineName in here correct? Especially if these things get deployed to ACA or some other K8s environment.

Reply:
Also, I like the machine name because it is a good reminder that things like pod anti-affinity exist for improving isolation and performance of a heavily loaded system.